packages/trpc/server/routers/viewer/teams/create.handler.ts

```diff
@@ -16,11 +16,41 @@ type CreateOptions = {
 };
 
 export const createHandler = async ({ ctx, input }: CreateOptions) => {
+  const { user } = ctx;
   const { slug, name, logo } = input;
+  const isOrgChildTeam = !!user.organizationId;
+  let org;
+
+  // For orgs we want to create teams under the org
+  if (user.organizationId) {
+    org = await prisma.team.findFirst({
+      where: {
+        id: user.organizationId,
+      },
+      select: {
+        members: true,
+      },
+    });
+
+    if (!org) throw new TRPCError({ code: "NOT_FOUND" });
+
+    // Check if the user has permission to create a team under the org
+    if (
+      !org?.members.some(
+        (member) =>
+          (member.userId === user.id && member.role === MembershipRole.OWNER) ||
+          member.role === MembershipRole.ADMIN
+      )
+    ) {
```

Comment:
NIT: I think we can move some of this into the query itself ? Might make it a tiny bit faster

Reply:
```suggestion
    // Look if userId is owner/admin of team org
    const isOwnerAdminOfOrg = await ctx.prisma.membership.findFirst({
      where: {
        userId: ctx.user.id,
        teamId: childTeam.parentId,
        role: {
          in: ["OWNER", "ADMIN"],
        },
        accepted: true,
      },
    });
```
Something I used on in trpc router middleware
https://github.com/calcom/cal.com/pull/9551/files#diff-37aef050083ed9d17c03e39a0361b3d7782783311f576e46e1620e2cb2f5ee8a

Reply:
Thinking about this more, I can send the `isOrgAdmin` property from the frontend, whether that's from the session or the `me` query to the endpoint.

Reply:
@joeauyeung we never want to rely on frontend validation for this - a user could always modify the request to say they are an admin. We want to always check this permission on the backend